main.js

```diff
@@ -0,0 +1,278 @@
+console.log("let's code");
+
+let weatherComponent;
+
+/**
+ * Helper functions
+ */
+const APIKEY = "deecee58f4daa55a503c09ae97c1d3ab";
+
+const fetchData = async function (url) {
+  const resp = await fetch(url);
+  const data = await resp.json();
+  return data;
+}
+
+const getLatLonData = async function (cityName) {
+  const url = `http://api.openweathermap.org/geo/1.0/direct?q=${cityName}&limit=1&appid=${APIKEY}`;
+  const data = await fetchData(url);
+
+  // If location is not found
+  if(data.length === 0) return -1;
+
+  const {name, lat, lon} = data[0];
+  return {name, lat, lon};
+}; 
+
+const getCurrentWeatherData = async function (lon, lat, unit="imperial") {
+  const returnData = {}
+  const url = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=${unit}&appid=${APIKEY}`
+  const data = await fetchData(url);
+  returnData.name = data.name;
+  returnData.temp = data.main.temp;
```

Comment:
if by any reason, main is not being returned, this will crash. Safer to use ?
```suggestion
  returnData.temp = data.main?.temp;
```